ligo/src/base_DAO.mligo

```diff
@@ -3,77 +3,45 @@
- * to be transferred.
+ * Entrypoints that require the no xtz to be sent.
  * This checks for such xtz condition.
- *
- * Note: entrypoint parameters are packed as-is except for "GetVotePermitCounter"
- * and 'Vote`, see 'startup.mligo' for more info.
  *)
-let requiring_no_xtz (param : forbid_xtz_params) : (string * bytes) =
+let requiring_no_xtz (param, store, config : forbid_xtz_params * storage * config)
+    : operation list * storage =
   // check for no xtz
   if Tezos.amount > 0tez
-  then (failwith("FORBIDDEN_XTZ") : (string * bytes))
+  then (failwith("FORBIDDEN_XTZ") : return)
   else
     match param with
-    | Call_FA2 p -> fa2_eps p
-    | Drop_proposal p -> ("drop_proposal", Bytes.pack p)
-    | Transfer_ownership p -> ("transfer_ownership", Bytes.pack p)
-    | Accept_ownership p -> ("accept_ownership", Bytes.pack p)
-    | Migrate p -> ("migrate", Bytes.pack p)
-    | Confirm_migration p -> ("confirm_migration", Bytes.pack p)
-    | Propose p -> ("propose", Bytes.pack p)
-    | Vote p -> ("vote", Bytes.pack (p, Tezos.self_address))
-    | Set_voting_period p -> ("set_voting_period", Bytes.pack p)
-    | Set_quorum_threshold p -> ("set_quorum_threshold", Bytes.pack p)
-    | Flush p -> ("flush", Bytes.pack p)
-    | Burn p -> ("burn", Bytes.pack p)
-    | Mint p -> ("mint", Bytes.pack p)
-    | GetVotePermitCounter p ->
-        let callback_address = Tezos.address(p.callback)
-        in ("get_vote_permit_counter", Bytes.pack (p.param, callback_address))
-    | Get_total_supply p ->
-        let callback_address = Tezos.address(p.callback)
-        in ("get_total_supply", Bytes.pack (p.token_id, callback_address))
+    | Call_FA2 (p) -> call_fa2(p, store)
+    | Drop_proposal (p) -> drop_proposal(p, config, store)
+    | Transfer_ownership (p) -> transfer_ownership(p, store)
+    | Accept_ownership (p) -> accept_ownership(p, store)
+    | Migrate (p) -> migrate(p, store)
+    | Confirm_migration (p) -> confirm_migration(p, store)
+    | Propose (p) -> propose(p, config, store)
+    | Vote (p) -> vote(p, config, store)
+    | Set_voting_period (p) -> set_voting_period(p, config, store)
+    | Set_quorum_threshold (p) -> set_quorum_threshold(p, config, store)
+    | Flush (p) -> flush (p, config, store)
+    | Burn (p) -> burn(p, store)
+    | Mint (p) -> mint(p, store)
+    | GetVotePermitCounter (p) -> get_vote_permit_counter(p, store)
+    | Get_total_supply (p) -> get_total_supply(p, store)
 
 (*
- * Returns name and packed parameter of a stored entrypoint that requires the
- * contract to not be migrated.
- * This checks for such migration condition.
+ * Entrypoints that require the contract to be migrated.
+ * Note: This checks for such migration condition.
  *)
-let requiring_no_migration (param, store : migratable_parameter * storage) : (string * bytes) =
+let requiring_no_migration (param, store, config : migratable_parameter * storage * config)
```

Comment:
Perhaps we could use `full_storage` here and in `requiring_no_xtz`.

Reply:
It doesn't really make a difference in terms of compiled result, however not carrying `full_storage` allows us to avoid splitting it in both function (which would be necessary) and only do that in `base_DAO_contract`:
```ocaml
  let store : storage = full_store.0 in
  let config : config = full_store.1 in
```
which removes some boilerplate.

Reply:
Understandable, then let's leave it as is.